ui/src/utils/insertionSearch.ts

```diff
@@ -0,0 +1,18 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureNode } from '../treeDataProvider/AzureNode';
+
+// returns the index of where the node should be inserted
+export function insertionSearch(node: AzureNode, array: AzureNode[]): number {
+
+    // tslint:disable-next-line:no-increment-decrement
+    for (let i: number = 0; i < array.length; i++) {

```

Comment:
I think we should discourage putting stuff in 'utils' whenever possible (otherwise it becomes a dumping ground). No strong opinion, but I don't think this for-loop warrants its own file

Reply:
It felt a little messy to include it in the AzureParentNode.ts file to me as well though.  Where would you put it??

Reply:
How about leaving it in the same folder as AzureParentNode.ts?

Reply:
It's only 6 lines of code - why not put it directly in addNodeToCache?

Reply:
Oh, wow, then agreed, unless you plan to test it separately.